CRM/Contribute/Form/Contribution.php

```diff
@@ -462,6 +462,25 @@ public function setDefaultValues() {
       $defaults['campaign_id'] = $this->_pledgeValues['campaign_id'];
     }
 
+    if ($this->_id) {
+      $financialTrxn = $this->getlatestPayments();
+      if ($financialTrxn) {
+        $paymentProcessorID = CRM_Utils_Array::value('financial_trxn_id.payment_processor_id', $financialTrxn);
+        $result = civicrm_api3('FinancialTrxn', 'getsingle', array(
+          'return' => array("card_type", "pan_truncation"),
+          'id' => $financialTrxn['financial_trxn_id'],
+        ));
+        $defaults['card_type'] = CRM_Utils_Array::value('card_type', $result);
+        $defaults['pan_truncation'] = CRM_Utils_Array::value('pan_truncation', $result);
+
+        if ($paymentProcessorID) {
```

Comment:
@pradpnayak please reply to this comment indicating that you have accepted the point and will make a change or why you don't think it is necessary.